src/base/NekRSSeparateDomainProblem.C

```diff
@@ -0,0 +1,343 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2021 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#include "NekRSSeparateDomainProblem.h"
+#include "Moose.h"
+#include "AuxiliarySystem.h"
+#include "TimeStepper.h"
+#include "NekInterface.h"
+#include "TimedPrint.h"
+#include "MooseUtils.h"
+#include "CardinalUtils.h"
+
+#include "nekrs.hpp"
+#include "nekInterface/nekInterfaceAdapter.hpp"
+
+registerMooseObject("CardinalApp", NekRSSeparateDomainProblem);
+
+bool NekRSSeparateDomainProblem::_first = true;
+
+InputParameters
+NekRSSeparateDomainProblem::validParams()
+{
+  InputParameters params = NekRSProblemBase::validParams();
+  params.addParam<bool>("minimize_transfers_in", false, "Whether to only synchronize nekRS "
+    "for the direction TO_EXTERNAL_APP on multiapp synchronization steps");
+  params.addParam<bool>("minimize_transfers_out", false, "Whether to only synchronize nekRS "
+    "for the direction FROM_EXTERNAL_APP on multiapp synchronization steps");
+  params.addParam<bool>("moving_mesh", false, "Moving mesh");
```

Comment:
If setting `moving_mesh = true` will never do anything in this class, we can just remove it as an input parameter altogether.

Reply:
There's a few things I still left in the updated version of this file because I wasn't sure how they're used, like `minimize_transfers_in`. But along these lines, I tried removing `moving_mesh` from the problem type but I couldn't compile the problem without it. To be honest I didn't check into the reason behind this very much.